packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -1863,6 +1923,20 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
       this.validateSamlAuthenticationOptions(props.fineGrainedAccessControl?.samlAuthenticationOptions);
     }
 
+    if (props.capacity?.nodeOptions) {
+      // Validate coordinator node configuration
+      const coordinatorConfig = props.capacity.nodeOptions.find(opt => opt.nodeType === NodeType.COORDINATOR)?.nodeConfig;
+      if (coordinatorConfig?.enabled) {
+        const coordinatorType = initializeInstanceType(defaultCoordinatorInstanceType, coordinatorConfig.type);
+        if (!cdk.Token.isUnresolved(coordinatorType) && !coordinatorType.endsWith('.search')) {
```

Comment:
is the unresolved token check needed as `initializeInstanceType` has done the check.

Reply:
Followed the same as it was for warm, data, master. Can you please check